Feat logs

Sly-Agustin opened the pull request:
Added logs in different parts of the code

Sly-Agustin committed:
Implented health endpoint

- Implemented health endpoint
- created a controller
- Added routes
- created server.js
- Added packages to package.json for execution
- updated gitignore

Sly-Agustin committed:
Implemented basic create

- Implemented basic dream creation, there is no validation ATM, but inserts in online MongoDB fine if parameters are passed correctly.
- Added controller for dream creation
- Modified routes to include the new method

Sly-Agustin committed:
Update healthController.js

- Now health route succeed if the controller connects with the DB. It also send an error 500 if it fails to connect.

Sly-Agustin committed:
Added validation to schema

Sly-Agustin committed:
Update dream.js

- Added some specific validation fields

Sly-Agustin committed:
Validation

- Added validation rules on schema
- Added validation on the controller. It handles invalid input for all fields

Sly-Agustin committed:
Update createQuestionnaireController.js

Deleted comments

Sly-Agustin committed:
Added students

- Added students, they have a studentId that is autoincremented in MongoDB.
- Added a controller, router, and routes for student creation
- Tweaks on server.js for student

Sly-Agustin committed:
Updated controller

- Updated the getDreamControllerById, now it can fetch existing questionnaires. Validation still not implented for this.
- Renamed routes

Sly-Agustin committed:
Refactor

- Replaced a lot of "dreams" with "questionnaire"

Sly-Agustin committed:
Refactor createQuestionnaire

- Refactor createQuestionnaireController.js

Sly-Agustin committed:
Refactor

- Reimplemented getDreamControllerById
- The app connects to the server as soon as it start and not everytime it does a request
- Deleted console logs

Sly-Agustin committed:
Test getById

- Added mock test for getById
- Removed exec() from getById controller

Sly-Agustin committed:
Added test for create a questionnaire

Sly-Agustin committed:
Fixes required by pull request #1

- Changed the way the id is parsed, this is due to inconsistiencies in how parseInt works internally. Also a special case with empty spaces is considered because javascript interpret it as a 0, so isNaN would fail to catch it.
- Health controller now checks the connection based on  the actual readyState
- Removed redundant connection on createStudentController

Sly-Agustin committed:
Fixed createQuestionnaire test

- Test is mocked correctly this time

Sly-Agustin committed:
Added student validation

- createStudentController now validates the data it receives to create a new one

Sly-Agustin committed:
Update README.md

Sly-Agustin committed:
Updated documentation

- Updated docs with the .nvmrc file

Sly-Agustin committed:
Basic logs

- Installed winston for log management
- Implemented basic logs in controllers

Sly-Agustin committed:
Added different files for logs

- Added new loggers for different type of logs.
- Updated the controllers due to changes in loggers

mjuan99 commented on src/controller/createQuestionnaireController.js:
```diff
@@ -29,10 +30,12 @@ const createQuestionnaireController = async (req, res) => {
 
       const newQuestionnaire = await createQuestionnaire(Questionnaire, body);
       if(newQuestionnaire.errors){
+        logger.info('Questionnaire not created due to incorrect data')
```
It can be useful to log the incorrect data that produced the error (maybe at 'debug' level)

mjuan99 commented on src/controller/createQuestionnaireController.js:
```diff
@@ -29,10 +30,12 @@ const createQuestionnaireController = async (req, res) => {
       }
       else {
         await newQuestionnaire.save();
+        logger.info('Questionnaire created')
```
You can add the time that took to answer the request in all the response's logs. `process.hrtime.bigint()` can be useful for that purpose

mjuan99 commented on src/controller/students/createStudentController.js:
```diff
@@ -17,10 +18,12 @@ export const createStudentController = async (req, res) => {
 
         const newStundent = await createStudent(Student, body)
         if(newStundent.error){
+            studentLogger.info('Student not created due to incorrect data received');
```
It can be useful to log the incorrect data that produced the error (maybe at 'debug' level)

Sly-Agustin committed:
Changed loggers

- Changed the way the loggers format. I left some "simple" formats there to see if they are worth using for debug. However right now there are just json output debuggers

Sly-Agustin committed:
Changed to json loggs

- Logs are now in json instead of simple format
- Added process time to the log
